doc/API.md

```diff
@@ -368,6 +369,30 @@ disregarded for forward compatibility.
 None; the `result` field will be set to the empty object `{}`. Any value should be
 disregarded for forward compatibility.
 
+### `gethistory`
+
+#### Request
+
+| Field    | Type         | Description                                                          |
+| -------- | ------------ | -------------------------------------------------------------------- |
+| `cursor` | int          | Timestamp of the date to retrieve events after                       |
+| `limit`  | int          | Number of events to retrieve                                         |
+| `kind`   | string array | Type of the events to retrieve, can be `deposit`, `cancel`, `spend`  |
+
+#### Response
+
+| Field          | Type   | Description                                |
+| -------------- | ------ | ------------------------------------------ |
+| `events`       | array  | Array of [Event resource](#event-resource) |
+
+##### Event Resource
+
+| Field    | Type   | Description                                                                              |
+| -------- | ------ | ---------------------------------------------------------------------------------------- |
+| `kind`   | string | Type of the event, can be `deposit`, `cancel`, `spend`                                   |
+| `date`   | int    | Timestamp of the event                                                                   |
+| `amount` | int    | Outgoing or incoming Amount of the event transaction, `null` if the event is a `cancel`  |
+| `fee`    | int    | Fee of the event transaction, `null` if the event is a `deposit`                         |
```

Comment:
I think `amount` should always be there, as a total. And the `fee` be an optional part of it. For instance:
- For a deposit of 1M sat, `amount: 1_000_000`, `fee: null`
- For a cancel of 1M sat, `amount: `1_000_000`, `fee: 1000`
- For a spend of 1M sat, `amount: 1_000_000`, `fee: 2000`

Maybe for the spend it's awkward to have the `amount` not be the "spent amount" but i believe it makes sense to put the entire amount the previously-deposited vault.

Reply:
No I think it is confusing, notably for the cancel: what does amount means ? only fees are outgoing 

For the spend: do you count the spend output in the amount and the history has an other row with the change output amount as deposit ?

Reply:
It is difficult for the user to understand the move in wallet, maybe we can think of something like:
| kind    | fee  | amount   |
| ------  | ---- | -------  |
| cancel  | 1000 |          |
| spend   | 1000 | -1999000 |
| deposit |      | +2000000 |

Reply:
> what does amount means?

`amount` is the value of a vault.

From an accountancy point of view it seemed odd to me that you would have a deposit then a spend event that would not be of the same amount. I also dislike having nullable fields so reducing their number would be nice.

> [...] what about the change value?

Arg. Good point. Ok, i think your solution is the only thing left :)

Reply:
or we can seperate amount by `moved amount`, `incoming_amount` and `outgoing amount` and no null value only 0
| kind    | fee  | amount moved | incoming amount | outgoing_amount |
| ------  | ---- | ------------ | --------------  | --------------- |
| cancel  | 1000 | 100000       | 0               | 0               |
| spend   | 1000 | 2000000      | 0               | 1989000         |
| deposit |    0 | 2000000      | 2000000         | 0               |

In this case, the user can deduce that a change output was created by the spend: `moved amount > outgoing_amount + fee`

Reply:
Or sorry maybe moved amount can only be amount already in vault and we can call also incoming amount the change output amount:
in this case we have:
| kind    | fee  | amount moved | incoming amount | outgoing_amount |
| ------  | ---- | ------------ | --------------  | --------------- |
| cancel  | 1000 | 100000       | 99000           | 0               |
| spend   | 1000 | 2000000      | 10000           | 1989000         |
| deposit |    0 | 0            | 2000000         | 0               |

Reply:
I don't think that the accountancy tab should have a concept of what really happens onchain (movements). It should rather abstracts it to "we sent X-Y" rather than "we moved X and got Y back".

Reply:
I think the last level of abstraction can be done by the client, I think some other usecase will use this method, like overall statistics of volume of funds moving.}
Maybe one day we will have a new feature "consolidation" with a new line:
| kind    | fee  | amount moved | incoming amount | outgoing_amount |
| ------  | ---- | ------------ | --------------  | --------------- |
| cancel  | 1000 | 100000       | 99000           | 0               |
| spend   | 1000 | 2000000      | 10000           | 1989000         |
| deposit |    0 | 0            | 2000000         | 0               |
| consolidation | 3000 | 250000  | 247000| 0 |

Reply:
I think it's clearer if you just use `amount` with positive and negative values instead of incoming and outcoming, as you said:

kind | fee | amount
-- | -- | --
cancel | 1000 |  -0
spend | 1000 | -1999000
deposit | 0 | +2000000

Substituting the null value with "-0" (or "+0"? I don't know)
And maybe putting a negative number under "fee"? Like:

kind | fee | amount
-- | -- | --
cancel | -1000 |  -0
spend | -1000 | -1999000
deposit | -0 | +2000000

Reply:
I like the negative idea, but not for the fees. Conceptually it's weird to me to have a "negative fee" (you are paid for using it?)

Reply:
OK https://github.com/revault/revaultd/pull/198#issuecomment-831081469
let's go for unsigned value, we can deduce from the `kind` if it is incoming or leaving

I wanted to keep the two columns to have information about change output for a spend, I wanted that GUI display both columns but aggregate like you do in the column `amount` and when a user click on it, he gets the detail with amount of fund moved, change amount etc

Reply:
I would keep the CLI clean, even if it's just for us, and avoid having both the "amount" and "incoming/outcoming" as it's redundant. We can have an option of gethistory which displays incoming and outcoming instead of amount, like:

`revault-cli gethistory`:\

kind | fee | amount
-- | -- | --
cancel | 1000 | -0
spend | 1000 | -1999000
deposit | 0 | +2000000


`revault-cli gethistory -v` (or something better, idk)

kind | fee | incoming amount | outgoing_amount
-- | -- | -- | --
cancel | 1000 | 99000 | 0
spend | 1000 | 10000 | 1989000
deposit | 0 | 0  | 0

I'm not sure if we should show in the GUI how much for the change etc, is it useful for the user? 
Also, does the cancel have an "incoming amount"? It should be 0?

Reply:
Yes, you are right it is confusing for the cancel if there is no `amount moved`
Agree for:
| kind    | fee  | amount   |
| ------  | ---- | -------  |
| cancel  | 1000 |      0    |
| spend   | 1000 | 1999000 |
| deposit |    0  | 2000000 |

Reply:
I would say with "+" and "-", no?


kind | fee | amount
-- | -- | --
cancel | 1000 | 0
spend | 1000 | -1999000
deposit | 0 | +2000000